src/compiler/interface.jl

```diff
@@ -174,7 +174,7 @@ function pullback(f, ps::Params)
     for p in ps
       cache(cx)[p] = nothing
     end
-    back(Δ)
+    differential2legacy(back(legacy2differential(Δ)))
```

Comment:
Any idea what's going on here? It looks to me like the output of this line isn't actually getting used anywhere, but maybe there's something non-local going on? If so, would be good to comment on it.

Reply:
I agree, it would be great if there were comments everywhere explaining why things are being done. 😄

_I think_ it accumulates the gradients w.r.t. to the parameters, but I am not sure. There are no warnings/errors if the line is left as is, but removing it breaks things.